core/src/main/java/bisq/core/btc/nodes/LocalBitcoinNode.java

```diff
@@ -24,40 +53,321 @@
+        NioClient client = new NioClient(serverAddress, peer, connectionTimeout);
+
+        return client;
+    }
+
+    private static Level silence(Class klass) {
+        var logger = getLogger(klass);
+        var originalLevel = logger.getLevel();
+        logger.setLevel(Level.OFF);
+        return originalLevel;
+    }
+
+    private static void restoreLoggerLevel(Class klass, Level originalLevel) {
+        getLogger(klass).setLevel(originalLevel);
+    }
+
+    private static ch.qos.logback.classic.Logger getLogger(Class klass) {
+        return (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(klass);
+    }
+
+    private ListenableFuture<VersionMessage> getVersionMessage(Peer peer) {
+        SettableFuture<VersionMessage> peerVersionMessageFuture = SettableFuture.create();
+
+        var versionHandshakeDone = peer.getVersionHandshakeFuture();
+        FutureCallback<Peer> fetchPeerVersionMessage = new FutureCallback<Peer>() {
+            public void onSuccess(Peer peer) {
+                peerVersionMessageFuture.set(peer.getPeerVersionMessage());
+            }
+            public void onFailure(Throwable thr) {
+                // No action
+            }
+        };
+        Futures.addCallback(
+                versionHandshakeDone,
+                fetchPeerVersionMessage
+        );
+
+        PeerDisconnectedEventListener cancelIfConnectionFails =
+                new PeerDisconnectedEventListener() {
+                    public void onPeerDisconnected(Peer peer, int peerCount) {
+                        var peerVersionMessageAlreadyReceived =
+                                peerVersionMessageFuture.isDone();
+                        if (peerVersionMessageAlreadyReceived) {
+                            // This method is called whether or not the handshake was
+                            // successful. In case it was successful, we don't want to do
+                            // anything here.
+                            return;
+                        }
+                        // In some cases Peer will self-disconnect after receiving
+                        // node's VersionMessage, but before completing the handshake.
+                        // In such a case, we want to retrieve the VersionMessage.
+                        var peerVersionMessage = peer.getPeerVersionMessage();
+                        if (peerVersionMessage != null) {
+                            log.info("Handshake attempt was interrupted;"
+                                    + " however, the local node's version message was coerced.");
+                            peerVersionMessageFuture.set(peerVersionMessage);
+                        } else {
+                            log.info("Handshake attempt did not result in a version message exchange.");
+                            var mayInterruptWhileRunning = true;
+                            peerVersionMessageFuture.cancel(mayInterruptWhileRunning);
```

Comment:
Here again: This is something that helps only someone not using code editors like IntelliJ.

Reply:
Is this a win/lose situation where we have to adopt to the majority? It does help me, as I'm not using IntelliJ. Does it disturb IntelliJ users?

Reply:
I've commited this and other IDE-related changes, since Github back-and-forths can be slow, but I'd like an answer to this.

Reply:
I agree with @ripcurlx here. In general I think the code should try to stick to the current style of the project. I did not choose this style but I follow it to make it easier for everyone to read the code. If everyone were to follow their own preferred style it would make the code much less readable.

This kind of extra variable comes as a surprise here since it's not following the common style and it makes me wonder why it's done in this particular way, as if it has some significance.

Reply:
@sqrrm the reason why I use these is to not have "anonymous" values in calls. Example: `buildHouse(false)` v. `buildHouse(withRoof)`.